proposals/vocabularies-adr.md

```diff
@@ -0,0 +1,74 @@
+# [short title of solved problem and solution]
+
+* Status: proposed
+* Deciders: @gregsdennis, @jdesrosiers
+* Date: 2024-06-10
+
+Technical Story:
+
+- Issues discussing feature - https://github.com/json-schema-org/json-schema-spec/issues?q=is%3Aopen+is%3Aissue+label%3Avocabulary
+- ADR to extract from the spec and use feature life cycle - https://github.com/json-schema-org/json-schema-spec/pull/1510
+
+## Context and Problem Statement
+
+The current approach to extending JSON Schema by providing custom keywords is
+very implementation-specific and therefore not interoperable.
+
+To address this deficiency, this document proposes vocabularies as a concept
+and a new Core keyword, `$vocabulary` to support it.
+
+## Decision Drivers <!-- optional -->
+
+- Language-agnostic
+- Ease of use
+- Ease of implementation
+
+## Considered Options
+
+### Current design as included in 2019-09 and 2020-12.
+
+Vocabularies are external documents that describe how new keywords function.
+They can be in a specification style, or a blog post, or some other format.
```

Comment:
That's fair.  I'll massage this.

Reply:
The vocabs that I've created (and the OpenAPI one) are definitely in the specification style, but I remember discussing it with Henry, and he mentioned that it didn't always have to be.  It just had to be defined somewhere people who wanted to implement it had access to.  That could be a blog post or an internal wiki or whatever.

I think that allows for wider participation in defining new keywords than writing a spec-like document would because most people wouldn't bother with the precision.